### DISCUSSION TITLE
musikcube 0.97.0

### DISCUSSION THREAD

**Discussion by clangen** [clangen/musikcube#500] [General]
**important**: this release provides binaries generated with our newly re-written build and packaging system. while there are no functional changes to the app itself, there are quite a few quality of life changes for non-Windows platforms:

1. all `macOS` and `Linux` builds (including `armhf` builds for the `Raspberry Pi`) are completely self-contained, and no longer need additional dependencies installed to run. 
2. all `Linux` builds are distributed with `ffmpeg-5.0` libraries, the latest version as of the time of this writing.
3. `Raspberry Pi` builds now have `rpm` and `tar.bz2` packages, in addition `deb` archives.
4. `macOS` binaries no longer require `homebrew` to run, and also include `ffmpeg-5.0`

if you're still building from source (or `homebrew`) there's no need to worry, the process is still exactly the same as it's always been!

here are some quick links to the binaries provided below:

* `Windows`
  * [win64](#)
  * [win32](#)
  * [win32 with milkdrop](#)
* `macOS`
  * [intel (x86_64)](#)
  * [m1 (arm64)](#)
* `Linux x86_64`
  * [deb](#)
  * [rpm](#)
  * [tarball](#)
* `Linux armhf (Raspberry Pi)`
  * [deb](#)
  * [rpm](#)
  * [tarball](#)

<hr /><em>This discussion was created from the release <a href='https://github.com/clangen/musikcube/releases/tag/0.97.0'>musikcube 0.97.0</a>.</em>

**Comment by clangen** [comment]
Unfortunately `musikcube` doesn't have officially supported packages for most mainstream Linux distributions, and on `macOS` it requires `homebrew` to function properly.

This is not only an inconvenience to end-users, but it's also an inconvenience to me, as it has historically taken a minimum of 5 hours to generate packages for all the platforms we support, and then an additional couple hours to test them for each release. Even still, many users experience problems with certain dependencies like `boost`, which change regularly and are often not backwards compatible, causing `musikcube` to stop functioning until it's re-compiled or re-installed.

To address these issues, I've been working behind the scenes on a new method for building and distributing binaries. The new system works by packaging all required dependencies (`boost`, `libcurl`, `openssl`, `ffmpeg`, etc) and distributing them with `musikcube`. The result is a much smaller set of packages that I can generate much more quickly. Additionally, the packages should generally "just work" for most users, most of the time, without requiring installation of additional packages (or in the case of `macOS`, entire package mangers).

Another benefit is that the binaries will be much less likely to break during operating system updates.

Release version `0.97.0` (and hopefully all subsequent releases) will only provide these new packages, but the repository will still support the original packaging scripts, just in case anyone finds them useful.

**If you have a few minutes please respond to this post with your operating system and version, and whether or not the new packages work for you I'd greatly appreciate it.**

Thanks!

**Reply by ElectronicBattle** [reply]
I installed this 0.97 .deb for use with my Linux Mint 20.3 "Ulyana" (Release amd64 20200624) and it works perfectly.

**Comment by Travisyard** [comment]
All successful:

- Standalone x86_64 deb package installed in Debian 11 LXC
- Windows x86_64 build
- Building it from source the same way as I always have on Armbian Buster (aarch64)
